### DISCUSSION TITLE
Getting error about URL / Cant see Database > Connection?

### DISCUSSION THREAD

**Discussion by Daniel-PT** [tillsteinbach/VWsFriend#654] [Q&A]
Hi all

I have now installed 
vwsfrined-grafana and
vwsfriend
and made a postgresql in my postgresql docker.

Can i can test the connection with this uri

postgresql://vwsfriend:PW@192.168.1.50:5432/vwsfriend

But i dont see the "Database" in the top?

![image](https://github.com/user-attachments/assets/e28ae1b6-9ab5-4eac-b7be-94fc051a6b20)

![image](https://github.com/user-attachments/assets/60c61377-fab8-4ac9-af51-cd53a37d1350)

IF i install it woth docker desktop on a Windows VM i looks fine.

![image](https://github.com/user-attachments/assets/f576c693-314b-4bab-b77b-e50d9528f10d)

Is it the variables that is wroung in unraid?

![image](https://github.com/user-attachments/assets/11ede966-2bbe-4317-bb13-e3f50aad2ab1)

![image](https://github.com/user-attachments/assets/955e1559-248f-4d32-906a-7173d61c0013)

![image](https://github.com/user-attachments/assets/ca358bf9-7227-4a99-bdc6-4435bf2db30b)

Regards Daniel and thanks

**Comment by Daniel-PT** [comment]
@tillsteinbach  can you meaby help? :)

**Reply by Mirabilia-cyber** [reply]
I have the same issue, I can see only the Status and Logout in the VWsFriend. In vwsfriend-grafana I have the same exact errors. I installed the three images in docker, following the configuration file with all the variables, and using postgres as the DB. In the database none of the tables has been created.

**Reply by Mirabilia-cyber** [reply]
@Daniel-PT I found the issue for me. In the vwsfriend container creation command I added the "--with-database" parameter, so from:

`/bin/sh -c 'vwsfriend --username=${VWSFRIEND_USERNAME} --password=${VWSFRIEND_PASSWORD} --weconnect-username=${WECONNECT_USER} --weconnect-password=${WECONNECT_PASSWORD} --weconnect-spin=${WECONNECT_SPIN} --interval=${WECONNECT_INTERVAL} --port=${VWSFRIEND_PORT} --database-url=${DATABASE_URL} --config-dir=/config ${ADDITIONAL_PARAMETERS}'`

it should be changed as the following:

`/bin/sh -c 'vwsfriend --with-database --username=${VWSFRIEND_USERNAME} --password=${VWSFRIEND_PASSWORD} --weconnect-username=${WECONNECT_USER} --weconnect-password=${WECONNECT_PASSWORD} --weconnect-spin=${WECONNECT_SPIN} --interval=${WECONNECT_INTERVAL} --port=${VWSFRIEND_PORT} --database-url=${DATABASE_URL} --config-dir=/config ${ADDITIONAL_PARAMETERS}'`

Hope this helps.

**Reply by Daniel-PT** [reply]
> @Daniel-PT I found the issue for me. In the vwsfriend container creation command I added the "--with-database" parameter, so from:
> 
> `/bin/sh -c 'vwsfriend --username=${VWSFRIEND_USERNAME} --password=${VWSFRIEND_PASSWORD} --weconnect-username=${WECONNECT_USER} --weconnect-password=${WECONNECT_PASSWORD} --weconnect-spin=${WECONNECT_SPIN} --interval=${WECONNECT_INTERVAL} --port=${VWSFRIEND_PORT} --database-url=${DATABASE_URL} --config-dir=/config ${ADDITIONAL_PARAMETERS}'`
> 
> it should be changed as the following:
> 
> `/bin/sh -c 'vwsfriend --with-database --username=${VWSFRIEND_USERNAME} --password=${VWSFRIEND_PASSWORD} --weconnect-username=${WECONNECT_USER} --weconnect-password=${WECONNECT_PASSWORD} --weconnect-spin=${WECONNECT_SPIN} --interval=${WECONNECT_INTERVAL} --port=${VWSFRIEND_PORT} --database-url=${DATABASE_URL} --config-dir=/config ${ADDITIONAL_PARAMETERS}'`
> 
> Hope this helps.

Hi Mirabilia
Meny meny thanks for getting back! So nice to hear you made it working!

But can you tell me a little more about that? I dont fully understadn the "creation command" ? :)
I have just added the vwsfriend from the "docker hub" in the "apps" tab in unraid :)

**Reply by Mirabilia-cyber** [reply]
I'm using the QNAP NAS, so it might be different for you, but basically for me it works like this:

I pulled 3 images, one for postgres, one for vwsfriend and one for vwfriend-grafana.

Each image is being used by a docker container, and each docker container has a configuration file/setup. 

In these files I added my variables (username/password for the API, username/password for vwsfriend and the postgres db/username/password/IP-address).

The setup file of the container for the vwsfriend included the command I pasted above to start the vwsfriend webapp, so I modified that to have the creation the tables and use the postgres DB.

Hope this helps.

**Reply by Daniel-PT** [reply]
Many thanks!

I was told to try the "Post arguments" in Unraid. But it dosent work :(

![image](https://github.com/user-attachments/assets/5932c37b-f4a3-4c89-9b71-644b4fbbf2b4)

**Reply by Mirabilia-cyber** [reply]
Just a note: the webui variable has still the [IP] to be changed to the IP of your vwsfriend container. 
Do you have a configuration file to edit before creating the vwsfriend docker container?

**Reply by Daniel-PT** [reply]
Okay thanks! :) I have just edit the webui to the ip:4000 :)
No becurse it "pulls" it from docker-hub. Like it is "guessing" what the right variables are :(

**Reply by Mirabilia-cyber** [reply]
Where do you insert the variables? In the installation guide it's mentioned that there should be yml and env files to edit. Not sure if in your case you can edit them after the container has been pulled.
https://github.com/tillsteinbach/VWsFriend?tab=readme-ov-file#how-to-start

**Reply by Daniel-PT** [reply]
In my first screenshot of this post you can see that Unraid is trying to "guess" the variables from docker-hubt.
But i cant find the env files. And yea that is strange. I really cant get my head around it :(

**Comment by Daniel-PT** [comment]
Wow i got help from chatgtp :D
I could see in the logs that there was and error
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.InsufficientPrivilege) permission denied for schema public

So after i did some grant permissions in the postgress container i now have "database"
![image](https://github.com/user-attachments/assets/1ae42078-6f70-421f-9605-da9c821f8856)

Wow!
